### PR TITLE
Homework 7 task 2

### PR DESCRIPTION
**Pull request by ivanzuev78**

### CODE REVIEW

File: HomeWork_7/hw2.py [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
+    """
 
+    def type_this_string(string_to_type):
+        real_typed_string = ""
+        for char in string_to_type:
+            if char == "#":
+                real_typed_string = real_typed_string[:-1]
+            else:
+                real_typed_string += char
+            print(real_typed_string)
```
Comment (Cjkjvfnby) [inline HomeWork_7/hw2.py:30]: Linter definitely should catch this.  Please check your settings.  Looks like you disable it in a much broader scope than needed.

File: HomeWork_7/hw2.py [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-    Output: False
-    Explanation: s becomes "c" while t becomes "b".
+        Input: a = "a#c", t = "b"
+        Output: False
+        Explanation: s becomes "c" while t becomes "b".
 
-"""
+    """
 
+    def type_this_string(string_to_type):
```
Comment (Cjkjvfnby) [inline HomeWork_7/hw2.py:23]: What is the algorithmic complexity of this function?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork_7/hw2.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork_7/hw2.py:23]: Didn't think about it. I suppose it is O(n).

Comment (Cjkjvfnby) [reply HomeWork_7/hw2.py:23]: Can it be better?

Comment (ivanzuev78) [reply HomeWork_7/hw2.py:23]: It should be ok now.

File: HomeWork_7/hw2.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_7/hw2.py:30]: Yes, you're right. I fixed it.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please check which files are added to the commit.  There are to many of them.

File: HomeWork_7/hw2.py [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
+        chars_in_real_typed_string = []
+        for char in string_to_type:
+            if char == "#" and chars_in_real_typed_string:
+                chars_in_real_typed_string.pop(-1)
```
Comment (Cjkjvfnby) [inline HomeWork_7/hw2.py:27]: Please check `pop` signature.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**